app/controllers/projects/samples/metadata/fields_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Projects
+  module Samples
+    module Metadata
+      # Controller actions for Project Samples Metadata Fields Controller
+      class FieldsController < Projects::Samples::ApplicationController
+        respond_to :turbo_stream
+
+        # Param is received as:
+        # params: {sample: {edit_field: {key: {old_key: new_key}, value: {old_value: new_value}}}
+        # Fields that have not been changed will have equal old and new
+        def update # rubocop:disable Metrics/AbcSize
+          authorize! @project, to: :update_sample?
+          updated_metadata_field = ::Samples::Metadata::Fields::EditService.new(@project, @sample, current_user,
+                                                                                edit_field_params).execute
+          if @sample.errors.any?
+            render status: :unprocessable_entity,
+                   locals: { key: edit_field_params['edit_field']['key'].keys[0],
+                             value: edit_field_params['edit_field']['value'].keys[0] }
+          else
+            render_params = get_render_status_and_message(updated_metadata_field)
+            render status: render_params[:status], locals: { type: render_params[:message][:type],
+                                                             message: render_params[:message][:message] }
+          end
+        end
+
+        private
+
+        def edit_field_params
+          params.require(:sample).permit(edit_field: {})
```

Comment:
**nitpick (blocking):** You should be able to update this params method to be specific, i.e. use the keys present in comment above `update` method. So just incorporating `key` and `value` as being a hash should provide more context and then if bad keys were given it wouldn't try to pass them.

Reply:
Hopefully I didn't misinterpret what you meant - Added in a0e2712c46389d63c119a6251e2dab1bc003c610